server/src/main/java/org/opensearch/index/mapper/MapperBuilderSettings.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.index.mapper;
+
+/**
+ * An interface that defines settings for MapperBuilder implementations.
+ *
+ * @opensearch.experimental
+ */
+public interface MapperBuilderSettings {
```

Comment:
nit : Settings is a bit confusing , as it compares against actual 'settings'.

MapperBuilderProperties or something similar can work